pkg/core/object/fmt.go

```diff
@@ -149,25 +143,16 @@ func (v *FormatValidator) validate(obj *object.Object, unprepared, isParent bool
 	switch obj.Type() {
 	case object.TypeStorageGroup: //nolint:staticcheck // TypeStorageGroup is deprecated and that's exactly what we want to check here.
 		return fmt.Errorf("strorage group type is no longer supported")
-	case object.TypeTombstone:
-		if !unprepared && version.SysObjTargetShouldBeInHeader(obj.Version()) {
-			if len(obj.Payload()) > 0 {
-				return errors.New("system object has payload")
-			}
-			if obj.AssociatedObject().IsZero() {
-				return errors.New("system object has zero associated object")
-			}
-		}
-	case object.TypeLock:
+	case object.TypeLock, object.TypeTombstone:
 		if !unprepared {
 			if !version.SysObjTargetShouldBeInHeader(obj.Version()) {
-				return errors.New("obsolete LOCK object version")
+				return fmt.Errorf("obsolete %s object version", obj.Type())
```

Comment:
can this check be simplified to just a version check? this is literally what this PR does: no TS/LOCK with version 2.18- are not allowed (but in both cases of `unprepared`)...